dev/CONTRIBUTING.md

```diff
@@ -100,87 +94,60 @@ Run `builder run check` before committing to ensure lint and tests are passing.
+
+### Flexibility
+
+Victory is designed to be as flexible as possible. Code should be free of hard-coded values or one-off solutions. The following patterns also help keep the library flexible: 
+
+**Seperating Rendered Components**
+
+Any rendered component (_i.e._ `<line/>`, `<text/>` and even `<g/>`) should be written as seperate components, and included via `defaultProps` so that they can be overridden the users. This pattern also allows us to use the same code base to support a react native version of Victory. 
+
+These primitive rendered elements are kept in the `victory-core` repo. Please check [here](https://github.com/FormidableLabs/victory-core/tree/master/src/victory-primitives) first before writing a new rendered component. Rendered components should be stateless and as general as possible. [`Point`](https://github.com/FormidableLabs/victory-core/blob/master/src/victory-primitives/point.js) is a good example of a rendered component. Notice that `Point` is written so that it can easily be extended to a [react native compatible version](https://github.com/FormidableLabs/victory-core-native/blob/master/lib/components/victory-primitives/point.js). The appropriate versions of `Point` can then be included as `defaultProps` in [`VictoryScatter`](https://github.com/FormidableLabs/victory-chart/blob/master/src/components/victory-scatter/victory-scatter.js#L353), and the [native version](https://github.com/FormidableLabs/victory-chart-native/blob/master/lib/components/victory-scatter.js)
+
+**Support Data Accessor Props**
+
+Rather than requiring a rigid data structure, Victory components should provide accessor props for formatting whatever data the user provides into a format the component is expecting. Use this [`createAccessor` function](https://github.com/FormidableLabs/victory-core/blob/master/src/victory-util/helpers.js#L117) to build an accessor from props and use it to format data. 
+
+**Support themes**
+
+`defaultProps` should not contain any layout related props because these values may be provided by themes. Necessary layout props should be provided via [`fallbackProps`](https://github.com/FormidableLabs/victory-chart/blob/master/src/components/victory-scatter/victory-scatter.js#L11) instead, and merged with props and themes as needed.
+
+### Sensible Defaults
+
+All Victory components should render _something_ even if no props are provided. This can be accomplished with default data and other `defaultProps`. Victory components should also have sensible behavior even when the minimum of props are provided. For example, if only `data` is provided, a component like `VictoryBar` should calculate a domain from the data provided. All Victory components should also use the default `grayscale` theme so that they have sensible styling even when no styles are provided by the user.
```

Comment:
maybe change 3rd sentence to -->
> For example, if only `data` is provided, `VictoryBar` [calculates a domain from the data provided](https://github.com/FormidableLabs/victory-chart/blob/master/src/components/victory-bar/victory-bar.js).

...and link to the best line to illustrate the point. (With my naivety, I'm not sure exactly what that would be)